galite-core/src/main/kotlin/org/kopi/galite/visual/ApplicationContext.kt

```diff
@@ -0,0 +1,363 @@
+     * @return The [Application] menu.
+     */
+    val menu: VMenuTree
+      get() = applicationContext!!.application.menu
+
+    /**
+     * Returns the [LocalizationManager] instance.
+     * @return The [LocalizationManager] instance.
+     */
+    val localizationManager: LocalizationManager
+      get() = applicationContext!!.application.localizationManager
+
+    /**
+     * Returns the default application [Locale].
+     * @return The default application [Locale].
+     */
+    val defaultLocale: Locale
+      get() = applicationContext!!.application.defaultLocale
+
+    /**
+     * Returns the application [Registry].
+     * @return the application [Registry].
+     */
+    val registry: Registry
+      get() = applicationContext!!.application.registry
+
+    /**
+     * Returns the application [DBContext].
+     * @return The application [DBContext].
+     */
+    val dBContext: DBContext
+      get() = applicationContext!!.application.dBContext
+
+    /**
+     * Returns `true` if the [Application] should only generate help.
+     * @return `true` if the [Application] should only generate help.
+     */
+    val isGeneratingHelp: Boolean
+      get() = applicationContext!!.application.isGeneratingHelp
+
+    /**
+     * Displays an error message outside a model context. This can happen when launching a module.
+     * @param parent The parent component.
+     * @param message The message to be displayed.
+     */
+    fun displayError(parent: UComponent, message: String) {
+      applicationContext!!.application.displayError(parent, message)
+    }
+
+    // ---------------------------------------------------------------------
+    // SEND A BUG REPORT
+    // ---------------------------------------------------------------------
+
+    /**
+     * Reports a trouble at execution time.
+     *
+     * @param     module          the module where the trouble was detected
+     * @param     reason          the exception that triggered the bug report
+     */
+    fun reportTrouble(module: String,
+                      place: String,
+                      data: String,
+                      reason: Throwable?) {
+      if (applicationContext!!.application.isNobugReport) {
+        println("notice: reporting trouble is disabled, no mail will be sent.")
+        System.err.println(reason!!.message)
+        reason.printStackTrace(System.err)
+      } else {
+        var revision: String? = null
+        var releaseDate: String? = null
+        val versionArray: Array<String> = Utils.getVersion()
+        for (i in versionArray.indices) {
+          if (versionArray[i].startsWith("Revision: ")) {
+            revision = versionArray[i].substring(10)
+          } else if (versionArray[i].startsWith("Last Changed Date: ")) {
+            releaseDate = versionArray[i].substring(19)
+          }
+        }
+        if (defaults == null) {
+          System.err.println("ERROR: No application configuration available")
```

Comment:
Use when instead of if 
Use versionArray.forEach{ instead of for (i in ...indices )